File: frontend/src/api/guide.ts
import client from './client';

export type GuideMode = 'guided' | 'collaborator' | 'expert' | 'identity_creation';

export interface GuideOption {
  label: string;
  value: string;
}

export interface IdentityDraft {
  name?: string;
  purpose?: string;
  tone?: string;
  platforms?: string[];
  communication_style?: string;
  content_limits?: string;
}

export interface GuideState {
  step: number;
  objective?: string;
  audience?: string;
  platform?: string;
  tone?: string;
  topics?: string[] | string;
  postsPerDay?: number | string;
  scheduleStrategy?: string;
  extra_context?: string;
  conversation_summary?: string;
  identity_draft?: IdentityDraft;
  identity_id?: string;
}

export interface GuideNextRequest {
  current_step: number;
  mode: GuideMode;
  state: GuideState;
  user_input?: string;
  user_value?: string;
  guide_session_id: string;
}

export interface GuideNextResponse {
  assistant_message: string;
  options: GuideOption[];
  next_step: number;
  state_patch: Partial<GuideState>;
  status?: 'success' | 'blocked' | 'error';
}

export const guideApi = {
  nextStep: async (request: GuideNextRequest): Promise<GuideNextResponse> => {
    console.log("➡️ [API] guide.nextStep REQUEST:", request);
    try {
      const response = await client.post('/guide/next', request);
      console.log("⬅️ [API] guide.nextStep RESPONSE:", response.data);
      return response.data;
    } catch (error) {
      console.error("❌ [API] guide.nextStep ERROR:", error);
      throw error;
    }
  }
};

File: backend/app/schemas/guide.py
from pydantic import BaseModel, field_validator
from typing import List, Optional, Any, Dict, Union
from enum import Enum

class GuideMode(str, Enum):
    GUIDED = "guided"
    COLLABORATOR = "collaborator"
    EXPERT = "expert"
    IDENTITY_CREATION = "identity_creation"

class GuideOption(BaseModel):
    label: str
    value: str

class UserProfile(BaseModel):
    profession: Optional[str] = None
    specialty: Optional[str] = None
    target_audience_profile: Optional[str] = None
    bio_summary: Optional[str] = None
    has_uploaded_cv: bool = False

class IdentityDraft(BaseModel):
    name: Optional[str] = None
    purpose: Optional[str] = None
    tone: Optional[str] = None
    platforms: Optional[List[str]] = None
    communication_style: Optional[str] = None
    content_limits: Optional[str] = None
    # MVP PRO Fields (kept for compatibility)
    identity_type: Optional[str] = None
    campaign_objective: Optional[str] = None
    target_audience: Optional[str] = None
    language: Optional[str] = None
    preferred_cta: Optional[str] = None
    frequency: Optional[str] = None

class GuideState(BaseModel):
    step: int
    user_profile: Optional[UserProfile] = None
    objective: Optional[Union[str, List[str]]] = None
    audience: Optional[Union[str, List[str]]] = None
    platform: Optional[Union[str, List[str]]] = None
    tone: Optional[Union[str, List[str]]] = None
    topics: Optional[Union[List[str], str]] = None
    postsPerDay: Optional[Union[int, str]] = None
    scheduleStrategy: Optional[str] = None
    extra_context: Optional[str] = None
    conversation_summary: Optional[str] = ""
    identity_draft: Optional[IdentityDraft] = None
    identity_id: Optional[str] = None

    @field_validator('objective', 'audience', 'platform', 'tone', mode='before')
    @classmethod
    def normalize_list_to_string(cls, v):
        if isinstance(v, list):
            return ", ".join(map(str, v))
        return v

class GuideNextRequest(BaseModel):
    current_step: int
    mode: GuideMode = GuideMode.GUIDED
    state: GuideState
    user_input: Optional[str] = None
    user_value: Optional[str] = None
    guide_session_id: Optional[str] = None # Correlation ID for logging

class GuideNextResponse(BaseModel):
    assistant_message: str
    options: List[GuideOption] = []
    next_step: int
    state_patch: Dict[str, Any]
    status: str = "success" # success, blocked, error
